apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/PackageNameProvider.kt

```diff
@@ -3,35 +3,75 @@ package com.apollographql.apollo.compiler
 import java.io.File
 
 class PackageNameProvider(
-    /**
-     * the root package name for all classes:
-     *
-     * fragments will use rootPackageName.fragment
-     * types will use rootPackageName.type
-     * operations, will use rootPackageName.{relativePosition to rootDir}
-     */
-    private val rootPackageName: String?,
-    private val rootDir: File?,
+    private val rootPackageName: String,
+
     /**
      * the package name for fragments and types will be written there
      */
-    val irPackageName: String,
+    val schemaPackageName: String,
+
     /**
      * the package name for fragments, types, queries and mutations
      * This will flatten all the classes in the same package name. If you have operations in subfolders,
      * use rootPackageName instead
      */
+    @Deprecated("")
```

Comment:
`outputPackageName` is the one that flattens everything (https://github.com/apollographql/apollo-android/issues/1367). `rootPackageName ` is better as it will just prefix the package name while preserving the structure.

Reply:
I just realized your message was certainly about the missing deprecation message. I'll add that.